trace/opencensus/trace/tracer/context_tracer.py

```diff
@@ -78,6 +78,21 @@ def set_enabled(self):
         else:
             return False
 
+    def trace_decorator(self):
+        """Decorator to trace a function."""
+
+        def decorator(func):
+
+            def wrapper(*args, **kwargs):
```

Comment:
I'm not sure this is exactly what we want.  This will trace every execution of the wrapped function in the same context.  It seems like the wrapper should fetch the trace context every time it is executed.